synapse/config/cas.py

```diff
@@ -27,13 +27,16 @@ def read_config(self, config):
         if cas_config:
             self.cas_enabled = True
             self.cas_server_url = cas_config["server_url"]
+            self.cas_user_id_prefix = cas_config["user_id_prefix"]
```

Comment:
I'd argue it's safer to be required with empty string supported but requires explicit enablement. That way, the likelihood of enabling CAS and that allowing someone else to log in as an already created user is reduced.